packages/frontend/src/features/post/components/PostList/PostList.tsx

```diff
@@ -126,22 +126,22 @@ export default function PostList() {
         post: PostInfo,
     ): Promise<boolean> => {
         try {
-            if (post.isMine) {
+            if (post.isMine && post.finalAction !== null) {
                 let cancelAction: VoteAction
                 if (post.finalAction === VoteAction.UPVOTE) {
                     cancelAction = VoteAction.CANCEL_UPVOTE
-                } else if (post.finalAction === VoteAction.DOWNVOTE) {
-                    cancelAction = VoteAction.CANCEL_DOWNVOTE
                 } else {
-                    throw new Error('Invalid finalAction')
+                    cancelAction = VoteAction.CANCEL_DOWNVOTE
                 }
 
-                await createVote({
-                    id,
-                    voteAction: cancelAction,
-                    votedNonce: post.votedNonce,
-                    votedEpoch: post.votedEpoch,
-                })
+                if (cancelAction) {
+                    await createVote({
+                        id,
+                        voteAction: cancelAction,
+                        votedNonce: post.votedNonce,
+                        votedEpoch: post.votedEpoch,
+                    })
+                }
```

Comment:
To enhance code readability, L130 - L144 can refactor to: 
```js
const cancelAction = post.finalAction === VoteAction.UPVOTE 
  ? VoteAction.CANCEL_UPVOTE
  : VoteAction.CANCEL_DOWNVOTE

await createVote({
  id,
  voteAction: cancelAction,
  votedNonce: post.votedNonce,
  votedEpoch: post.votedEpoch,
})
```

Reply:
fix it